model-server/src/main/kotlin/org/modelix/model/server/handlers/ModelReplicationServer.kt

```diff
@@ -103,18 +108,28 @@ class ModelReplicationServer(
     }
 
     override suspend fun PipelineContext<Unit, ApplicationCall>.getRepositories() {
-        call.respondText(repositoriesManager.getRepositories().joinToString("\n") { it.id })
+        call.respondText(
+            repositoriesManager.getRepositories()
+                .filter { call.hasPermission("repository", it.id, "list") }
+                .joinToString("\n") { it.id },
+        )
     }
 
     override suspend fun PipelineContext<Unit, ApplicationCall>.getRepositoryBranches(repository: String) {
-        call.respondText(repositoriesManager.getBranchNames(repositoryId(repository)).joinToString("\n"))
+        call.respondText(
+            repositoriesManager
+                .getBranchNames(repositoryId(repository))
+                .filter { call.hasPermission("repository", repository, "branch", it, "list") }
+                .joinToString("\n"),
+        )
     }
 
     override suspend fun PipelineContext<Unit, ApplicationCall>.getRepositoryBranchDelta(
```

Comment:
Missing check for newly added `getRepositoryBranchV1`.

Reply:
Added the permission check.